cache/edge_lambdas/src/redirector.ts

```diff
@@ -70,24 +70,39 @@ export const getRedirect = (
     return redirect301(host, literalRedirects[uriSansSlash]);
   }
 
-  if (request.querystring && queryRedirects[uriSansSlash]) {
-    const potentialRedirect = queryRedirects[uriSansSlash];
+  if (queryRedirects[uriSansSlash]) {
     const requestParams = new URLSearchParams(request.querystring);
-    // A redirect occurs if all of the params in the redirect rule are contained
-    // within the request
-    if (paramsAreSubset(requestParams, potentialRedirect.matchParams)) {
-      // Only forward params that are in `forwardParams`
-      const newParams = filterParams(
-        requestParams,
-        potentialRedirect.forwardParams
-      );
-      const requestParamsString = newParams.toString();
-      return redirect301(
-        host,
-        requestParamsString
-          ? potentialRedirect.redirectPath + '?' + requestParamsString
-          : potentialRedirect.redirectPath
-      );
+
+    // If the redirect has matchParams, pick the relevant one from the list
+    // Otherwise return the one that has none
+    const potentialRedirect = queryRedirects[uriSansSlash].find(q =>
+      q.matchParams
+        ? paramsAreSubset(requestParams, q.matchParams)
+        : !q.matchParams
+    );
+
+    if (potentialRedirect) {
+      // A redirect occurs if all of the params in the redirect rule (matchParams)
+      // are contained within the request, or if there are no matchParams
+      if (
+        (potentialRedirect.matchParams &&
+          paramsAreSubset(requestParams, potentialRedirect.matchParams)) ||
+        !potentialRedirect.matchParams
+      ) {
```

Comment:
I think this could be made a bit more explicit like...
```suggestion
      const requestMatchesAllParams = potentialRedirect.matchParams && paramsAreSubset(requestParams, potentialRedirect.matchParams);
      const noMatchParams = !potentialRedirect.matchParams;
      if (requestMatchesAllParams || noMatchParams) {
```

or alternatively

```suggestion
      if (paramsAreSubset(requestParams, potentialRedirect.matchParams ?? new URLSearchParams())) {
```

Reply:
I like the first one, I find the condition very readable and not sure the second one (and my one) is as much. Let's go for readability.